examples/azure/build-and-deploy-mender-artifact.yml

```diff
@@ -0,0 +1,110 @@
+#
+# The example shows how to build, upload and deploy a Mender Artifact.
+#
+# 'MENDER_SERVER_ACCESS_TOKEN' secret is required to be set in Azure DevOps pipeline settings
+#
+
+trigger:
+- main
+
+
+variables:
+- name: MENDER_SERVER_URL # Mender server URL
+  value: https://hosted.mender.io
+- name: MENDER_ARTIFACT_NAME # Mender Artifact file name
+  value: artifact.mender
+- name: MENDER_RELEASE_NAME # The name of Mender release for the Mender Artifact
+  value: release-1.0
+- name: MENDER_DEPLOYMENT_NAME # The name of Mender deployment
+  value: deployment-release-1.0
+- name: MENDER_DEVICES_GROUP_NAME # # The name of Mender devices group a deployment with Mender Artifact will be triggered to
+  value: group-dev
+
+
+resources:
+  # Use other repositories for templates
+  # https://learn.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops#use-other-repositories
+  repositories:
+    - repository: templates
+      type: github
+      name: mendersoftware/mender-ci-workflows
+      ref: refs/heads/master
+      # GitHub service connection
+      # https://learn.microsoft.com/en-us/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml#github-service-connection
+      endpoint: github_service_connection
+
+
+#
+# Builds Mender Artifacts using 'mender-artifact' tool
+#
+jobs:
+- job: build_mender_artifact
+  pool:
+    vmImage: ubuntu-latest
+  # conrainer with pre-installed mender-artifact and mender-cli tools
+  container: mendersoftware/mender-ci-tools:mender-master
+  steps:
+  # Creates Mender Artifact, using randomly generated dummy name
+  # In this example, hello-mender.py represents a single file Python application that we have
+  # built and test in our CI and now we want to deploy into a device running Mender. We will
+  # package it as a Single File Application Update. For more information visit Mender docs:
+  # https://docs.mender.io/artifact-creation/create-an-artifact
+  - script: |
+      curl -O https://raw.githubusercontent.com/mendersoftware/mender-ci-workflows/master/examples/hello-mender.py
+      chmod 755 hello-mender.py
+      single-file-artifact-gen \
+        --device-type raspberrypi4 \
+        -o ${MENDER_ARTIFACT_NAME} \
+        -n ${MENDER_RELEASE_NAME} \
+        --software-name hello-mender \
+        --software-version 1.0 \
+        --dest-dir /usr/local/bin \
+        hello-mender.py
+    displayName: Build Mender Artifact
+  # Uploads Azure DevOps artifacts to Azure artifact storage to pass them to the further jobs
+  - task: PublishPipelineArtifact@1
+    inputs:
+      targetPath: $(MENDER_ARTIFACT_NAME)
+      artifactName: mender
```

Comment:
Maybe `mender` can be ambiguous? We can use something more explicit like `mender_artifact` or something.